python/ray/util/sgd/torch/worker_group.py

```diff
@@ -423,8 +440,19 @@ def shutdown(self, force=False):
         # Remove worker placement group.
         if self._worker_placement_group:
             remove_placement_group(self._worker_placement_group)
+            removed_placement_group = self._worker_placement_group
             self._worker_placement_group = None
 
+            def is_placement_group_removed():
+                table = ray.util.placement_group_table(removed_placement_group)
+                if "state" not in table:
+                    return False
+                return table["state"] == "REMOVED"
+
+            # Wait the placement_group been deleted
+            wait_for_condition(is_placement_group_removed)
```

Comment:
Can you do something like this instead:

```
success = wait_for_condition(is_placement_group_removed, timeout=SGD_PLACEMENT_GROUP_TIMEOUT_S)
if not success:
    logger.warning(f"Placement Group removal is not successful after {SGD_PLACEMENT_GROUP_TIMEOUT_S} seconds.")
```

The reason being even if placement group removal is unsuccessful, we can still continue training, just with a fewer number of workers.